### PR TITLE
enhancement(kubernetes platform): Support exposing internal mertics out of the box in our Helm charts

### LINKED ISSUE #3799: Add an option to expose internal metrics via Prometheus to the Helm Chart
A lot of apps that are deployed to Kubernetes integrate into the Prometheus metrics collection pipelines out of the box. This works through an automated discovery process performed by the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator), and all the app has to do is expose a Prometheus endpoint and set some annotations, opting in for collection.

Vector has internal metrics, and we could very easily expose them via Prometheus endpoint. So, pretty much, all we have to do is add a proper knob to our Helm Chart. We can even enable it by default.

This should be a significant quality of life improvement for people already using Prometheus for metrics collection, because without any extra effort they'll be getting important data points about vector operation itself into their monitoring/altering system.

### PR DESCRIPTION
**Pull request by MOZGIII**
This PR adds support for exposing `internal_metrics` via a `prometheus` sink in out Helm charts out of the box.
It also adds opt-in Prometheus annotations to `Pod` (`prometheus.io/scrape: "true"` and etc) and `prometheus-operator`-powered `PodMonitor` to simplify the integration into the existing cluster monitoring pipelines.

Closes #3799.

To do:

- [x] `prometheus` annotations
- [x] `prometheus-operator` `PodMonotor`s
- [x] E2E tests
- [x] fix race condition at E2E tests (https://github.com/timberio/vector/runs/1363029594?check_suite_focus=true)
- [ ] docs

### CODE REVIEW

File: distribution/helm/vector-aggregator/values.yaml [hunk 1]
```diff
@@ -173,6 +173,57 @@ vectorSource:
...
+  # Unless `excludeInternalMetrics` is set to `true`, in which case you're
+  # responsible of wiring up the internal metrics.
+  inputs: []
+  # Set this to `true` to opt-out from automatically adding the built-in
+  # internal metrics source to the inputs.
+  excludeInternalMetrics: false
+  # The address to listen at.
+  listenAddress: "0.0.0.0"
+  # The port to listen at.
+  listenPort: "8080"
```
Comment (eeyun) [inline distribution/helm/vector-aggregator/values.yaml:208]: Is this the default listen port for the prometheus sink or just what we have it set to? If it is the default then no worries. If it's not the default we should consider changing it. I feel like people use 8080 for dev and debug frequently.

Comment (MOZGIII) [reply distribution/helm/vector-aggregator/values.yaml:208]: It's just the one I picked. I agree, let's change it! What's the port we'd like to use here? How about `9090`?

**Review by eeyun — APPROVED** [review]

File: distribution/helm/vector-aggregator/values.yaml (hunk 1 above)
Comment (eeyun) [reply distribution/helm/vector-aggregator/values.yaml:208]: 9090 sounds perfect! Otherwise the PR looks great!

**Pull request merged**